addons/stock/res_config_view.xml

```diff
@@ -119,5 +123,19 @@
         <menuitem id="menu_stock_config_settings" name="Warehouse" parent="base.menu_config"
             sequence="5" action="action_stock_config_settings"/>
 
+        <record id="stock_negative_inherited_company_view" model="ir.ui.view">
+            <field name="name">res.company.stock.negative</field>
+            <field name="model">res.company</field>
+            <field name="priority">24</field>
+            <field name="inherit_id" ref="base.view_company_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//group[@name='account_grp']" position="after">
+                    <group name="stock_config" string="Warehouse">
+                        <field name="no_negative_stock"/>
+                    </group>
+                </xpath>
+            </field>
+        </record>
+
     </data>
 </openerp>
```

Comment:
why inherited company view in config file ? better move it in separate xml file. res_company_view.xml.